CHANGELOG.md

```diff
@@ -1,5 +1,12 @@
 # UNRELEASED
 
+**Other changes**
+
+-   Add `OlmMachine.markAllTrackedUsersAsDirty` to invalidate the device lists
+    for all known users. This is required for [MSC4186](https://github.com/matrix-org/matrix-spec-proposals/pull/4186)
+    clients as the server may give up trying to persist device list updates for
+    the client at some point, after which the client must treat all devices as dirty.
```

Comment:
could you also document the matrix-rust-sdk update, including any changes that the update may have caused and may be relevant to the application. See below for examples of the convention.

(And yes, this is a massive faff. Please join my campaign for proper, documented, releases of matrix-rust-sdk!)

Reply:
Uh, how should I do this? Literally look at the new commits on rust SDK?

Reply:
Pretty much. You might want to only look at commits that affect the crates we're using. And `matrix-sdk-crypto` and `matrix-sdk-indexeddb` have changelogs, so you could just look at the changes to their changelogs.

I use this very hacky script:

```
#!/bin/bash
#
# Usage: diff-for-wasm.sh <commit> [<commit>]

changelog_crates=(
    matrix-sdk-indexeddb
    matrix-sdk-crypto
)

other_crates=(
    matrix-sdk-common
    matrix-sdk-store-encryption
    matrix-sdk-qrcode
)

args=()
for crate in "${changelog_crates[@]}"; do
    args+=("./crates/${crate}/CHANGELOG.md")
done
for crate in "${other_crates[@]}"; do
    args+=("./crates/${crate}")
done
exec git diff "$@" "${args[@]}"
```

Reply:
Unsure how to use your script, but looking at https://github.com/matrix-org/matrix-rust-sdk/compare/07aa6d7bc...2408df8b there isn't anything worth mentioning afaict?

Reply:
Maybe something about https://github.com/matrix-org/matrix-rust-sdk/pull/3986 ? It doesn't seem to change the API though?

Reply:
Yeah agreed, there's nothing terribly important here. Thanks for checking. Could you just add a one-liner then:

```suggestion
-   Add `OlmMachine.markAllTrackedUsersAsDirty` to invalidate the device lists
    for all known users. This is required for [MSC4186](https://github.com/matrix-org/matrix-spec-proposals/pull/4186)
    clients as the server may give up trying to persist device list updates for
    the client at some point, after which the client must treat all devices as dirty.

-   Update matrix-rust-sdk to `2408df8bf`. No changes relevant to these bindings.
```

Reply:
in fact, I'll do that now